### PR TITLE
Add block damage abort event

### PR DESCRIPTION
**Pull request by jonasHanhan**
### Problem

Skript has no event for Paper's `BlockDamageAbortEvent`, so scripts cannot detect when a player stops damaging a block before it breaks.

### Solution

Add an `on block damage abort` event under the modern block module and register it through `SyntaxRegistry`.

The event exposes the documented player and item stack through `EventValueRegistry`; the inherited block event value remains available as `event-block`. A parser test covers all three values.

### Testing Completed

- `clean quickTest` on Paper 26.1.2 with JDK 25: 792 script tests succeeded, 0 failed, documentation tests passed
- `clean checkstyleMain`: build passed, with no Checkstyle violations in `BlockModule` or `EvtBlockDamageAbort`

### Supporting Information

The implementation is located in `org.skriptlang.skript.bukkit.block.elements.events` and uses the current syntax and event-value registries as requested in the issue discussion.

---
**Completes:** #8752
**Related:** none
**AI assistance:** OpenAI Codex was used to inspect project conventions, implement the event and registry wiring, add test coverage, and run and interpret the Gradle checks. I reviewed the final diff and test results before submission.

### CODE REVIEW

File: src/main/java/org/skriptlang/skript/bukkit/block/elements/events/EvtBlockDamageAbort.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			BukkitSyntaxInfos.Event.builder(EvtBlockDamageAbort.class, "Block Damage Abort")
+				.supplier(EvtBlockDamageAbort::new)
+				.addEvent(BlockDamageAbortEvent.class)
+				.addPattern("block damage abort")
+				.addDescription("Called when a player stops damaging a block before it breaks.")
+				.addExample("""
+					on block damage abort:
+						send "You stopped damaging %event-block% with %event-item%." to event-player
+					""")
+				.addSince("2.16")
```
Comment (AnOwlBe) [inline src/main/java/org/skriptlang/skript/bukkit/block/elements/events/EvtBlockDamageAbort.java]:
```suggestion
				.addSince("INSERT VERSION")
```

File: src/main/java/org/skriptlang/skript/bukkit/block/elements/events/EvtBlockDamageAbort.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import org.bukkit.event.block.BlockDamageAbortEvent;
+import org.bukkit.inventory.ItemStack;
+import org.skriptlang.skript.bukkit.lang.eventvalue.EventValue;
+import org.skriptlang.skript.bukkit.lang.eventvalue.EventValueRegistry;
+import org.skriptlang.skript.bukkit.registration.BukkitSyntaxInfos;
+import org.skriptlang.skript.registration.SyntaxRegistry;
+
+public class EvtBlockDamageAbort extends SimpleEvent {
+
+	public EvtBlockDamageAbort() {
```
Comment (AnOwlBe) [inline src/main/java/org/skriptlang/skript/bukkit/block/elements/events/EvtBlockDamageAbort.java]: this isn't needed I don't think the (public EvtBlockDamageAbort() { with the super)

File: src/main/java/org/skriptlang/skript/bukkit/block/elements/events/EvtBlockDamageAbort.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package org.skriptlang.skript.bukkit.block.elements.events;
```
Comment (AnOwlBe) [inline src/main/java/org/skriptlang/skript/bukkit/block/elements/events/EvtBlockDamageAbort.java]: Since this whole event has nothing special theres no need for it to be its own file - hence why I would've suggested it goes in BlockEvents but that isn't in dev/feature yet but its definitely something to note since it'll probably need to  be moved there once it is in dev/feature

**Review by AnOwlBe — CHANGES_REQUESTED** [review]
few things

File: src/main/java/org/skriptlang/skript/bukkit/block/BlockModule.java [hunk 4]
```diff
@@ -24,7 +26,10 @@ public Iterable<AddonModule> children() {
 
 	@Override
 	public void loadSelf(SkriptAddon addon) {
-		// intentionally left blank
+		EventValueRegistry eventValueRegistry = addon.registry(EventValueRegistry.class);
+		register(addon,
+			syntaxRegistry -> EvtBlockDamageAbort.register(syntaxRegistry, eventValueRegistry)
```
Comment (AnOwlBe) [inline src/main/java/org/skriptlang/skript/bukkit/block/BlockModule.java]: theres no need for this to be inside a register(addon since its only a single thing simply instead can be 1 line of `EvtBlockDamageAbort.register(syntaxRegistry, eventValueRegistry);`

**Comment by sovdeeth** [comment]
You need to actually test your changes manually ingame! the tests passing here mean very little since none of the event's logic is actually tested.

File: src/main/java/org/skriptlang/skript/bukkit/block/BlockModule.java [hunk 5]
```diff
@@ -27,9 +33,32 @@ public Iterable<AddonModule> children() {
...
+				.addExample("""
+					on block damage abort:
+						send "You stopped damaging %event-block% with %event-item%." to event-player
+					""")
+				.addSince("INSERT VERSION")
+				.build()
 		);
+
+		eventValueRegistry.register(EventValue.simple(
+			BlockDamageAbortEvent.class,
```
Comment (AnOwlBe) [inline src/main/java/org/skriptlang/skript/bukkit/block/BlockModule.java]: I don't see a reason for these to  be split up into 5 lines

I think the cleanest option is to just use the none simple event-value builder with a .getter()

File: src/main/java/org/skriptlang/skript/bukkit/block/BlockModule.java [hunk 6]
```diff
@@ -27,9 +33,32 @@ public Iterable<AddonModule> children() {
 	@Override
 	public void loadSelf(SkriptAddon addon) {
 		EventValueRegistry eventValueRegistry = addon.registry(EventValueRegistry.class);
-		register(addon,
-			syntaxRegistry -> EvtBlockDamageAbort.register(syntaxRegistry, eventValueRegistry)
+		SyntaxRegistry syntaxRegistry = moduleRegistry(addon);
+
+		syntaxRegistry.register(
```
Comment (AnOwlBe) [inline src/main/java/org/skriptlang/skript/bukkit/block/BlockModule.java:38]: this event needs a `supplier` with a SimpleEvent("block damage abort") (so proper toString can be handled for the event instead of it just showing `simple event`

**Review by AnOwlBe — CHANGES_REQUESTED** [review]

File: src/main/java/org/skriptlang/skript/bukkit/block/BlockModule.java (hunk 6 above)
Comment (jonasHanhan) [reply src/main/java/org/skriptlang/skript/bukkit/block/BlockModule.java:38]: Addressed in fdb6addc by adding a `SimpleEvent("block damage abort")` supplier.

**Review by AnOwlBe — APPROVED** [review]
Do make sure you have tested this ingame  manually and that it works

and also make sure to run quicktest + junitquick to make sure they pass

**Comment by sovdeeth** [comment]
re: deleted comment. AbortProbe is not a real minecraft account. Please put down the ai for a moment, actually do some work, and test your PR in a real minecraft server. If you cannot do this, and cannot talk to us without using AI, we will be closing this PR. Talking to us as a human is basic courtesy.
